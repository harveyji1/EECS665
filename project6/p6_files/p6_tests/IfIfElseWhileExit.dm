func: () void{
    while(true){
        a: int = 1;
    }
    if(true){
        a: int = 1;
    }
    if(true){
        a: int = 1;
    } else {
        a: int = 1;
        a++;
        a--;
    }
}