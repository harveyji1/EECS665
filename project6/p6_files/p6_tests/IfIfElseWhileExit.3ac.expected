[BEGIN GLOBALS]
func
[END GLOBALS]
[BEGIN func LOCALS]
a (local var of 8 bytes)
a (local var of 8 bytes)
a (local var of 8 bytes)
a (local var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
lbl_1:      nop
            IFZ 1 GOTO lbl_2
            goto lbl_1
lbl_2:      nop
            IFZ 1 GOTO lbl_3
lbl_3:      nop
            IFZ 1 GOTO lbl_4
            goto lbl_5
lbl_4:      nop
            [a] := [a] ADD64 1
            [a] := [a] SUB64 1
lbl_5:      nop
lbl_0:      leave func

